lib/OpenQA/Scheduler/Scheduler.pm

```diff
@@ -1197,11 +1195,21 @@ sub job_schedule_iso {
         # As this comes in from an API call, URL will be URI-encoded
         # This obviously creates a vuln if untrusted users can POST ISOs
         $args{ISOURL} = url_unescape($args{ISOURL});
+        # set $args{ISO} to the URL filename if we only got ISOURL.
+        # This has to happen *before* _generate_jobs so the jobs have
+        # ISO set
+        if (!$args{ISO}) {
+            # this is slightly hackish but will work fine unless you
+            # have a pretty weird URI with query params or something
+            $args{ISO} = (split('/', $args{ISOURL}))[-1];
```

Comment:
I would prefer something like
```perl
use Mojo::URL;
$args{ISO} = Mojo::URL->new($args{ISOURL})->path->parts->[-1];
```
just to be safe and allow query params if needed.